### PR TITLE
Iceprog mpsse refactor

### PR DESCRIPTION
**Pull request by esden**
These changes move the MPSSE specific functions into it's own module. This should make the code bit less cluttered and also allow for the reuse of the MPSSE specific code for other projects.

I have changed the bank selection to be numeric instead of an enum so that we don't refer to any ftdi internal structures and api in the main header file, this was the only place we would need it otherwise.

### CODE REVIEW

File: iceprog/mpsse.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ *  ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ *  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#ifndef MPSSE_H
+#define MPSSE_H
+
+void mpsse_check_rx(void);
+void mpsse_error(int status);
+uint8_t mpsse_recv_byte(void);
```
Comment (daveshah1) [inline iceprog/mpsse.h]: This should include `stdint.h` to provide `uint8_t` so this can be a standalone header

File: iceprog/mpsse.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+uint8_t mpsse_xfer_spi_bits(uint8_t data, int n);
+void mpsse_set_gpio(uint8_t gpio, uint8_t direction);
+int mpsse_readb_low(void);
+int mpsse_readb_high(void);
+void mpsse_send_dummy_bytes(uint8_t n);
+void mpsse_send_dummy_bit(void);
+void mpsse_init(int ifnum, const char *devstr, bool slow_clock);
+void mpsse_close(void);
+
+#endif /* MPSSE_H */
```
Comment (daveshah1) [inline iceprog/mpsse.h]: Missing newline at end of file

**Comment by esden** [comment]
Thank you for merging. I was about to fix the things @daveshah1 pointed out. I will send a new cosmetic cleanup PR later. :)

**Comment by cliffordwolf** [comment]
Did that already in 15c9bfd. :)

**Comment by esden** [comment]
Ok, awesome! Thank you again for merging this. :D

**Pull request merged**